### DISCUSSION TITLE
Why is this recognized correctly but still the number is wrong?

### DISCUSSION THREAD

**Discussion by brandstaetter** [jomjol/AI-on-the-edge-device#2314] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/644820/232182225-e28fe28e-5d9d-40e4-9b18-0a9bc73b49b8.png)
![image](https://user-images.githubusercontent.com/644820/232182247-9cd8aa73-4839-45e9-b9e1-6263e152c645.png)
is there anything I can do to improve this? It's getting annoying, happens every time the first analog digit is between 0.8 and 0.99

**Comment by friedpa** [comment]
I cannot follow you... the Raw Value is correct, the last previous Value correctly read was 511.7699, now the rate between the two figures is too high -> 1.0922, as, I guess, your max rate value is much lower. 
Two way to fix it, set the Privious Value manually to the last Raw Value via Webinterface, or adjust it via MQTT automatically (PR#2252)

**Comment by brandstaetter** [comment]
The meter reads 511.8621. The recognition sees the digit that is displaying the `2` correctly as `1.9` but the value comes out as 512... I don't get the logic there.

The logic should be something like: 
```
if (least_significant_digit < round(least_significant_digit)) and (most_significant_analog > 0.7) 
then
 value = lower(least_significant_digit)
```

**Comment by friedpa** [comment]
Is my assumption now correct, that your meter shows in the third digital number, in the picture "2" too early in full, even if the read value shoud be just 511,8621 instead of 512,8621? If yes, did you try "Check Digit Increase Consistency = true"?

**Reply by ghost** [reply, marked as answer]
Check Digit Increase Consistency = true does nothing with newer models dig-cont / dig-class100.

May be "Analog/Digital Transition Start" set to 7.2 could help. Looks like the last digit starts the transition very early.

**Comment by brandstaetter** [comment]
Well, it is recognized as 1.9 regardless, so I don't understand where the rounding up happens; I just activated your suggested parameter and will report back if it improves the situation! Thanks so far already!

**Comment by friedpa** [comment]
@haverland: You are right, I am using still a very, very old model from the beginning of the project, which works still fine for me.

**Comment by brandstaetter** [comment]
Seems to work nearly perfect now, thanks!
